mlflow/utils/environment.py

```diff
@@ -729,6 +732,108 @@ def _get_pip_install_mlflow():
+        "Done adding requirements!\n\n"
+        f"Old requirements:\n{pformat([str(req) for req in old_reqs])}\n\n"
+        f"Updated requirements:\n{pformat(updated_reqs)}\n"
+    )
+
+
+def _remove_requirements_from_file(
+    reqs_to_remove: List[Requirement],
+    file_path: str,
+) -> None:
+    """
+    Removes requirements from the given requirements file (either `conda.yaml`
+    or `requirements.txt`). If any of the provided requirements are not present
+    in the file, they will be ignored. This function also ignores version specifiers.
+    For example, if the file contains "pandas==1.2.3", and the provided requirement
+    is "pandas", the function will remove "pandas==1.2.3" from the file.
+
+    Args:
+        reqs_to_remove (List[`Requirement`]): List of requirements to remove from the file
+        file_path (str): Path to the locally-saved requirements file. Must point to
+            either a `conda.yaml` or `requirements.txt` file.
+    """
+    old_reqs = _get_requirements_from_file(file_path)
+    updated_reqs = _remove_requirements(reqs_to_remove, old_reqs)
+    _write_requirements_to_file(file_path, updated_reqs)
+    _logger.info(
+        "Done removing requirements!\n\n"
+        f"Old requirements:\n{pformat([str(req) for req in old_reqs])}\n\n"
+        f"Updated requirements:\n{pformat(updated_reqs)}\n"
+    )
+
+
+def _get_requirements_from_file(
+    file_path: str,
+) -> List[Requirement]:
+    path = pathlib.Path(file_path)
+    data = path.read_text()
+    if path.name == _CONDA_ENV_FILE_NAME:
+        conda_env = yaml.safe_load(data)
+        reqs = _get_pip_deps(conda_env)
+    else:
+        reqs = data.splitlines()
+    return [Requirement(req) for req in reqs]
+
+
+def _write_requirements_to_file(
+    file_path: str,
+    new_reqs: List[str],
+) -> None:
+    path = pathlib.Path(file_path)
+    if path.name == _CONDA_ENV_FILE_NAME:
+        conda_env = yaml.safe_load(path.read_text())
+        conda_env = _overwrite_pip_deps(conda_env, new_reqs)
+        with path.open("w") as file:
+            yaml.dump(conda_env, file)
+    else:
+        path.write_text("\n".join(new_reqs))
+
+
+def _add_or_overwrite_requirements(
+    new_reqs: List[Requirement],
+    old_reqs: List[Requirement],
+) -> List[str]:
+    deduped_new_reqs = _deduplicate_requirements([str(req) for req in new_reqs])
+    deduped_new_reqs = [Requirement(req) for req in deduped_new_reqs]
+
+    old_reqs_dict = {req.name: str(req) for req in old_reqs}
+    new_reqs_dict = {req.name: str(req) for req in deduped_new_reqs}
+    old_reqs_dict.update(new_reqs_dict)
+    return list(old_reqs_dict.values())
+
+
+def _remove_requirements(
+    reqs_to_remove: List[Requirement],
+    old_reqs: List[Requirement],
+) -> List[str]:
+    old_reqs_dict = {req.name: str(req) for req in old_reqs}
+    for req in reqs_to_remove:
+        if req.name in old_reqs_dict:
+            _logger.info(f'"{req.name}" not found in requirements, ignoring')
```

Comment:
```suggestion
            _logger.warning(f'"{req.name}" not found in requirements, ignoring')
```

can we use `warning` here?